walrus-derive/src/lib.rs

```diff
@@ -30,45 +30,90 @@ pub fn walrus_expr(_attr: TokenStream, input: TokenStream) -> TokenStream {
     TokenStream::from(expanded)
 }
 
-fn get_enum_variants(input: &DeriveInput) -> Vec<syn::Variant> {
-    match &input.data {
-        syn::Data::Enum(en) => en.variants.iter().cloned().collect(),
+struct WalrusVariant {
+    syn: syn::Variant,
+    fields: Vec<WalrusFieldOpts>,
+}
+
+#[derive(Default)]
+struct WalrusFieldOpts {
+    skip_visit: bool,
```

Comment:
Have you seen https://crates.io/crates/darling btw?

It is pretty awesome for deriving your derives (or at least deriving parsing of derive options)

yo dawg

Reply:
This looks pretty slick! I'm having a bit of a tough time following it though. I think for now this is an internal-enough derive that we can be pretty lossy/strict/bad, but if we get too much more advanced let's switch to something like darling